reactioncommerce/reaction
# Fix #3754: check permission before publish

Resolves #3754  
Impact: **minor**  
Type: **bugfix**

## Issue
The shops to publish that were being sent from the client were not being checked for access.

## Solution
Added a check to check if the user has permission to see the shops that are being asked to publish.

## Testing
2. Login as admin, enable Marketplace, check checkbox to allow users to register as sellers
3. Logout and register a new account
4. Visit profile, click "Become a seller"
5. Observe no error and everything working.
